src/model/cloud-runner/state/cloud-runner-state.ts

```diff
@@ -0,0 +1,96 @@
+import path from 'path';
+import { BuildParameters } from '../..';
+import CloudRunnerEnvironmentVariable from '../services/cloud-runner-environment-variable';
+import CloudRunnerNamespace from '../services/cloud-runner-namespace';
+import { CloudRunnerProviderInterface } from '../services/cloud-runner-provider-interface';
+import CloudRunnerSecret from '../services/cloud-runner-secret';
+import { TaskParameterSerializer } from './task-parameter-serializer';
+
+export class CloudRunnerState {
+  static setup(buildParameters: BuildParameters) {
+    CloudRunnerState.buildParams = buildParameters;
+    if (CloudRunnerState.buildGuid === undefined) {
+      CloudRunnerState.buildGuid = CloudRunnerNamespace.generateBuildName(
+        CloudRunnerState.runNumber,
+        buildParameters.platform,
+      );
+    }
+    TaskParameterSerializer.setupDefaultSecrets();
+  }
+  public static CloudRunnerProviderPlatform: CloudRunnerProviderInterface;
+  public static buildParams: BuildParameters;
+  public static defaultSecrets: CloudRunnerSecret[];
+  public static buildGuid: string;
+  public static get branchName(): string {
+    return CloudRunnerState.buildParams.branch;
+  }
+  public static get buildPathFull(): string {
+    return path.join(`/`, CloudRunnerState.buildVolumeFolder, CloudRunnerState.buildGuid);
+  }
+  public static get builderPathFull(): string {
+    return path.join(CloudRunnerState.buildPathFull, `builder`);
```

Comment:
Prefer newlines between methods for diffing reasons.

Reply:
Is it possible to setup a linting rule @webbertakken ?